python/dgl/graph.py

```diff
@@ -765,7 +765,8 @@ class DGLGraph(DGLBaseGraph):
     edge_frame : FrameRef, optional
         Edge feature storage.
     multigraph : bool, optional
-        Whether the graph would be a multigraph (default: False)
+        Whether the graph would be a multigraph. If none, the flag will be determined
+        by scanning the whole graph. (default: None)
```

Comment:
this sounds very expensive. then we should force users to provide `multigraph` for graph store.

Reply:
please remove the default value here. https://github.com/dmlc/dgl/blob/master/python/dgl/contrib/graph_store.py#L1020

Reply:
For immutable graph, the flag will be only computed once and only when `is_multigraph` is called. That'll be fine?

Reply:
For giant graphs, I think it's better to ask users to provide this information? Is it fast to test it for a very large graph?

Reply:
Yeah, we still allow user to directly provide the flag. I guess in the giant graph example, it's better to specify that explicitly. What do you think?

Reply:
yes, this is what I meaned.